test/matlab_experimental/ctTestThermo.m

```diff
@@ -0,0 +1,482 @@
+            
+            self.checkMultiProperties('SP');
+            self.checkMultiProperties('SPX');
+            self.checkMultiProperties('SPY');
+            
+            self.checkMultiProperties('SV');
+            self.checkMultiProperties('SVX');
+            self.checkMultiProperties('SVY'); 
+
+            self.checkMultiProperties('DP');
+            self.checkMultiProperties('DPX');
+            self.checkMultiProperties('DPY'); 
+        end
+
+        % Check setter
+        function checkSetters(self, T1, D1, Y1)
+            val = self.phase.TDY;
+            T0 = val{1};
+            D0 = val{2};
+            Y0 = val{3};
+
+            self.phase.TDY = {T1, D1, Y1};
+            X1 = self.phase.X;
+            P1 = self.phase.P;
+            H1 = self.phase.H;
+            S1 = self.phase.S;
+            U1 = self.phase.U;
+            V1 = self.phase.V;
+
+            self.phase.TDY = {T0, D0, Y0};
+            self.phase.TPY = {T1, P1, Y1};
+            self.checkState(T1, D1, Y1);
+
+            self.phase.TDY = {T0, D0, Y0};
+            self.phase.UVY = {U1, V1, Y1};
+            self.checkState(T1, D1, Y1); 
+            
+            self.phase.TDY = {T0, D0, Y0};
+            self.phase.HPY = {H1, P1, Y1};
+            self.checkState(T1, D1, Y1);
+            
+            self.phase.TDY = {T0, D0, Y0};
+            self.phase.SPY = {S1, P1, Y1};
+            self.checkState(T1, D1, Y1);
+
+            self.phase.TDY = {T0, D0, Y0};
+            self.phase.TPX = {T1, P1, X1};
+            self.checkState(T1, D1, Y1); 
+            
+            self.phase.TDY = {T0, D0, Y0};
+            self.phase.UVX = {U1, V1, X1};
+            self.checkState(T1, D1, Y1);
+            
+            self.phase.TDY = {T0, D0, Y0};
+            self.phase.HPX = {H1, P1, X1};
+            self.checkState(T1, D1, Y1);
+            
+            self.phase.TDY = {T0, D0, Y0};
+            self.phase.SPX = {S1, P1, X1};
+            self.checkState(T1, D1, Y1);
+            
+            self.phase.TDY = {T0, D0, Y0};
+            self.phase.SVX = {S1, V1, X1};
+            self.checkState(T1, D1, Y1);
+            
+            self.phase.TDY = {T0, D0, Y0};
+            self.phase.SVY = {S1, V1, Y1};
+            self.checkState(T1, D1, Y1); 
+
+            self.phase.TDY = {T0, D0, Y0};
+            self.phase.DPX = {D1, P1, X1};
+            self.checkState(T1, D1, Y1);
+            
+            self.phase.TDY = {T0, D0, Y0};
+            self.phase.DPY = {D1, P1, Y1};
+            self.checkState(T1, D1, Y1); 
+        end
+    end
+
+    methods (Test)
+        % Test methods
+
+        function testBaseAttributes(self)
+            self.assumeInstanceOf(self.phase.solnName, ...
+                                      'char');
+            
+            % Missing method
+            % self.assumeInstanceOf(self.phase.phaseName, ...
+            %                           'char');
+            % 
```

Comment:
Rather than commented-out code (which can go unnoticed for a very long time), please use the `self.assumeFail()` method, which will always leave evidence behind in the test logs.